edb/server/compiler_pool/remote_worker.py

```diff
@@ -0,0 +1,481 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2022-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from __future__ import annotations
+from typing import *  # NoQA
+
+import argparse
+import gc
+import os
+import pickle
+import signal
+import time
+import traceback
+
+import immutables
+
+from edb import graphql
+
+from edb.common import debug
+from edb.common import devmode
+from edb.common import markup
+
+from edb.edgeql import parser as ql_parser
+
+from edb.pgsql import params as pgparams
+
+from edb.schema import schema as s_schema
+
+from edb.server import compiler
+from edb.server import config
+
+from . import amsg
+from . import state
+
+
+INITED: bool = False
+clients: immutables.Map[int, ClientSchema] = immutables.Map()
+BACKEND_RUNTIME_PARAMS: pgparams.BackendRuntimeParams = (
+    pgparams.get_default_runtime_params()
+)
+COMPILER: compiler.Compiler
+LAST_STATE: Optional[compiler.dbstate.CompilerConnectionState] = None
+STD_SCHEMA: s_schema.FlatSchema
```

Comment:
FWIW if you want to avoid code duplication we can move this state into a class (as well as `__sync__` and `__init_worker__` methods etc). At this stage you should know if the code is 99% similar or if it's different enough to warrant copy/paste.

Reply:
Ah yes - I'll likely re-organize the code with module-level functions, because the `compile_*` methods are largely different. Let me try now.